__tests__/app.test.js

```diff
@@ -129,6 +133,111 @@ describe("/api/articles/:article_id", () => {
+        .patch("/api/articles/1")
+        .send({ inc_votes: 6 })
+        .expect(200)
+        .then(() => {
+          return request(app)
+            .get("/api/articles/1")
+            .expect(200)
+            .then(({ body: { article } }) => {
+              expect(article).toMatchObject({
+                article_id: 1,
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 106,
+                article_img_url:
+                  "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+              });
+            });
+        });
+    });
+    test("PATCH 200: correctly decreases vote count", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({ inc_votes: -25 })
+        .expect(200)
+        .then(() => {
+          return request(app)
+            .get("/api/articles/1")
+            .expect(200)
+            .then(({ body: { article } }) => {
+              expect(article).toMatchObject({
+                article_id: 1,
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 75,
+                article_img_url:
+                  "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+              });
+            });
+        });
+    });
+    test("PATCH 404: responds not found if id is valid but not present", () => {
+      request(app)
+        .patch("/api/articles/8000")
+        .send({ inc_votes: 6 })
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Resource not found");
+        });
+    });
+    test("PATCH 400: responds bad request if id is invalid", () => {
+      request(app)
+        .patch("/api/articles/banana")
+        .send({ inc_votes: 6 })
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+    test("PATCH 400: responds bad request if no inc_votes key in payload", () => {
+      request(app)
+        .patch("/api/articles/3")
+        .send({ not_good_key: 6 })
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+    test("PATCH 400: responds bad request if inc_votes isn't an integer", () => {
+      request(app)
+        .patch("/api/articles/3")
+        .send({ inc_votes: "seven yay" })
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+  });
 });
```

Comment:
(This is Optional) but you may also want to do Status 200, if inc_votes is missing, should not update article and return original single article object (this may be included as a 400 but that could make for a brittle test suite)